apps/src/weblab/WebLab.js

```diff
@@ -242,6 +243,12 @@ WebLab.prototype.init = function (config) {
       />
     </Provider>
   ), document.getElementById(config.containerId));
+
+  window.onbeforeunload = evt => {
+    if (dashboard.project.hasProjectChanged()) {
```

Comment:
You can now import project instead of getting it from `window.dashboard` - they refer to the same object.  [Here's the exported `project` object](https://github.com/code-dot-org/code-dot-org/blob/4d4180033b5ca5235f663ca35db2175331e76ae4/apps/src/code-studio/initApp/project.js#L105) and [here's where we add it to dashboard](https://github.com/code-dot-org/code-dot-org/blob/4d4180033b5ca5235f663ca35db2175331e76ae4/apps/src/sites/studio/pages/code-studio.js#L49).

In fact, we already [import project in this file](https://github.com/code-dot-org/code-dot-org/blob/4d4180033b5ca5235f663ca35db2175331e76ae4/apps/src/weblab/WebLab.js#L15) and [use it this way at least once](https://github.com/code-dot-org/code-dot-org/blob/4d4180033b5ca5235f663ca35db2175331e76ae4/apps/src/weblab/WebLab.js#L272), although there are lots of other references to `window.dashboard.project` in here as well.